CustomLookup/CustomEntitySearch.cs

```diff
@@ -0,0 +1,187 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System.Net.Http.Formatting;
+using System.Text;
+using Microsoft.AspNetCore.Routing.Template;
+
+namespace CustomLookup
+{
+    /// <summary>
+    /// Based on sample custom skill provided in Azure Search. Provided a user-defined list of entities
+    /// this function determines the entities first occurrence within a given document. This list of entities
+    /// must repeatedly be provided by the user for each document.
```

Comment:
There's no way to supply per-document skill parameters in Cognitive Search pipeline that i can think of. Can you refactor this so that list of entities is a payload level entity rather than document level entity?

Reply:
Currently, I can't find a way to have the list of entities at a payload level as each text has a recordId

Reply:
Couldn't you use the output of entity detection, for example, be the input for your words parameter?

Reply:
In that case, all texts are under one recordId. Is that okay?

Reply:
If you mean the terms to look for, then yes, that is fine as one of the options, and since you added the configurable lookup file as an additional input, I think that would be fine.